ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/SandboxServer.scala

```diff
@@ -126,6 +126,14 @@ final class SandboxServer(
   def this(config: SandboxConfig, materializer: Materializer) =
     this(config, materializer, new Metrics(new MetricRegistry))
 
+  // NOTE(MH): We must do this _before_ we load the first package.
+  config.profileDir match {
+    case None => ()
+    case Some(profileDir) =>
+      Files.createDirectories(profileDir)
+      engine.startProfiling(profileDir)
+  }
```

Comment:
You can use `foreach` here.

```suggestion
  config.profileDir.foreach { profileDir =>
    Files.createDirectories(profileDir)
    engine.startProfiling(profileDir)
  }
```